### ISSUE TITLE
Cannot install 

### ISSUE THREAD

**Issue by JimHofman** [mwalle/rawsocket#5]
I'm unable to use pip install rawsocket.  I'm getting :
NOTE:  in the real display the ^ are pointing to ETH_P_ALL and AF_PACKET
                                                                            ^                         ^

Collecting rawsocket
  Using cached rawsocket-0.2.tar.gz (5.6 kB)
  Preparing metadata (setup.py) ... done
Using legacy 'setup.py install' for rawsocket, since package 'wheel' is not installed.
Installing collected packages: rawsocket
    Running setup.py install for rawsocket ... error
    ERROR: Command errored out with exit status 1:
     command: /Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-install-caf0jh69/rawsocket_635fd58588d4477bb8db33b61cd68911/setup.py'"'"'; __file__='"'"'/private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-install-caf0jh69/rawsocket_635fd58588d4477bb8db33b61cd68911/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-record-5p_nr033/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m/rawsocket
         cwd: /private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-install-caf0jh69/rawsocket_635fd58588d4477bb8db33b61cd68911/
    Complete output (14 lines):
    running install
    running build_helper
    building 'rawsocket-helper' helper
    creating build
    creating build/temp.macosx-10.9-x86_64-3.7
    cc -c rawsocket-helper.c -o build/temp.macosx-10.9-x86_64-3.7/rawsocket-helper.o
    rawsocket-helper.c:65:14: error: use of undeclared identifier 'ETH_P_ALL'
                    protocol = ETH_P_ALL;
                               ^
    rawsocket-helper.c:73:18: error: use of undeclared identifier 'AF_PACKET'
            raw_fd = socket(AF_PACKET, SOCK_RAW, htons(protocol));
                            ^
    2 errors generated.
    error: command 'cc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-install-caf0jh69/rawsocket_635fd58588d4477bb8db33b61cd68911/setup.py'"'"'; __file__='"'"'/private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-install-caf0jh69/rawsocket_635fd58588d4477bb8db33b61cd68911/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/7t/hdh8zd8j23v5hdjmnv_7fvmhpk6lyz/T/pip-record-5p_nr033/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m/rawsocket Check the logs for full command output.

**Comment by mwalle** [comment]
Hi,

is this macos? This library is linux only unfortunatly. As far as I know, only linux can pass the rights the it is done in this library.

**Comment by JimHofman** [comment]
Yes this is on a Mac. I should have noted that in the text above.  Thank you for the prompt response.
Do you happen to know of an equivalent module that works on MacOs?
 All the best to you and your team.

**Comment by mwalle** [comment]
I've just pushed a new version, which add some missing includes. please have a look at the latest master and see if it works for you. I see some traces of SCM_RIGHTS in the recvmsg man page of macos. So it might work or it might not work. but it should compile for you now.

**Comment by JimHofman** [comment]
Thank you Michael
